### PR TITLE
Dfs2 subsetting with physical coordinates

### PR DESCRIPTION
**Pull request by ecomodeller**
bild
Previously you had to find the idx first and then use .`isel()`. Now you can skip the first step.

### CODE REVIEW

**Comment by ecomodeller** [comment]
@simj-dhi is this what you need?
#604

**Comment by simj-dhi** [comment]
> @simj-dhi is this what you need? #604

Exactly! Awesome!

File: mikeio/dataset/_dataarray.py [hunk 1]
```diff
@@ -866,6 +870,27 @@ def sel(
...
+                    if k == "x":
+                        pos = 0
+                    if k == "y":
+                        pos = 1
+                
+                start = idx_start[pos][0] if idx_start is not None else None
+                stop = idx_stop[pos][0] if idx_stop is not None else None
+
+                idx = slice(start, stop)
+
```
Comment (jsmariegaard) [inline mikeio/dataset/_dataarray.py:890]: Maybe add a check that idx is not empty here? And throw nicer error message? 

In my testing I got this (which is not super bad, but could be better): 

ValueError                                Traceback (most recent call last)
/workspaces/mikeio/notebooks/Dfs2 - Various types.ipynb Cell 8 line 1
----> [1](vscode-notebook-cell://codespaces%2Bglorious-space-computing-machine-rp7pr4pq6rcx5p6/workspaces/mikeio/notebooks/Dfs2%20-%20Various%20types.ipynb#X56sdnNjb2RlLXJlbW90ZQ%3D%3D?line=0) da.sel(x=slice(0.4e6, 0.9e6), y=slice(6e7, 6.5e6))

File /workspaces/mikeio/mikeio/dataset/_dataarray.py:843, in DataArray.sel(self, time, **kwargs)
    738 """Return a new DataArray whose data is given by
    739 selecting index labels along the specified dimension(s).
    740 
   (...)
    840 geometry: Dfsu2D (3700 elements, 2090 nodes)
    841 """
    842 if any([isinstance(v, slice) for v in kwargs.values()]):
--> 843     return self._sel_with_slice(kwargs)
    845 da = self
    847 # select in space

File /workspaces/mikeio/mikeio/dataset/_dataarray.py:891, in DataArray._sel_with_slice(self, kwargs)
    887         stop = idx_stop[pos][0] if idx_stop is not None else None
    889         idx = slice(start, stop)
--> 891         self = self.isel(idx, axis=k)
    893 return self

File /workspaces/mikeio/mikeio/dataset/_dataarray.py:674, in DataArray.isel(self, idx, axis, **kwargs)
    672     idx = list(range(*idx.indices(self.shape[axis])))
...
    676     )  # TODO other option would be to have a NullDataArray
    678 idx = np.atleast_1d(idx)
    679 single_index = len(idx) == 1

ValueError: Empty index is not allowed

**Review by jsmariegaard — APPROVED** [review]
Great work - works as intended. 

Optional, add better error message if selection is empty (see comment)

File: mikeio/dataset/_dataarray.py (hunk 1 above)
Comment (ecomodeller) [reply mikeio/dataset/_dataarray.py:890]: Trying to understand the reason for the error, did you specify coordinates outside the domain and would like a helpful error message telling you this?

Comment (jsmariegaard) [reply mikeio/dataset/_dataarray.py:890]: yes, e.g. with UTM coordinates, accidentally swapping x and y, or selection too small to have any elements. But it's just a minor thing. Also fine to merge as is

**Pull request merged**